erpnext/stock/report/batch_wise_balance_history/batch_wise_balance_history.py

```diff
@@ -45,29 +45,30 @@ def get_columns(filters):
 
 
 def get_conditions(filters):
-	conditions = ""
+	conditions = []
 	if not filters.get("from_date"):
 		frappe.throw(_("'From Date' is required"))
 
 	if filters.get("to_date"):
-		conditions += " and posting_date <= '%s'" % filters["to_date"]
+		conditions.append("posting_date <= %(to_date)s")
 	else:
 		frappe.throw(_("'To Date' is required"))
 
-	return conditions
+	if filters.get("warehouse"):
+		filters.warehouse = frappe.parse_json(filters.get('warehouse'))
+		conditions.append("warehouse in %(warehouse)s")
```

Comment:
```suggestion
		conditions.append(" and warehouse in %(warehouse)s")
```

Reply:
@sahil28297 this is not a string. this is the list now just mention "and" gives error incorrect syntax thats why we need to join